client/src/main/java/com/paypal/selion/platform/grid/Grid.java

```diff
@@ -93,17 +98,10 @@ public static long getExecutionTimeoutValue() {
      *         {@link WebTest} flow.
      */
     public static RemoteWebDriver driver() {
-        // Need to throw an RuntimeException if already caught on SeleniumGridListener.beforeInvocation()
-        Exception exception = threadLocalException.get();
-        if (exception != null) {
-            if (exception instanceof RuntimeException) {
-                throw (RuntimeException) exception;
-            }
-            // convert the checked exception into a runtime exception.
-            throw new RuntimeException(exception.getMessage(), exception);
-        }
-
         AbstractTestSession testSession = getTestSession();
+        if (testSession == null) {
+            throw new IllegalStateException("Internal error. Test session not initialized.");
```

Comment:
Hmm.. I guess we can say something like "Make sure the SeLion SeleniumGridListener is enabled and you have annotated your test with `@MobileTest` or `@WebTest`...